lib/HLSL/DxilGenerationPass.cpp

```diff
@@ -313,7 +313,10 @@ class DxilGenerationPass : public ModulePass {
     // High-level metadata should now be turned into low-level metadata.
     const bool SkipInit = true;
     hlsl::DxilModule &DxilMod = M.GetOrCreateDxilModule(SkipInit);
-    auto pProps = &EntryPropsMap.begin()->second->props;
+    DxilFunctionProps *pProps = nullptr;
+    if (!SM->IsLib()) {
+      pProps = &EntryPropsMap.begin()->second->props;
```

Comment:
The change is good. I wonder if there's any reason why this code isn't doing things in this order:
```
InitDxilModuleFromHLModule
if (!SM->IsLib()) {
 DxilMod.SetShaderProperties(&EntryPropsMap.begin()->second->props);
}
DxilMod.ResetEntryPropsMap(std::move(EntryPropsMap));
```

Reply:
That won't work, because `SetShaderProperties` assumes the module already has a non-empty `m_DxilEntryPropsMap`, but this is the first materialization of a DxilModule, without initialization (meaning no loading metadata for these properties).  `ResetEntryPropsMap` will initialize that `m_DxilEntryPropsMap` by moving the contents from our local map, after which we can't get the same property pointer to call `SetShaderProperties`, which is necessary to set some extra state variables based on those properties for certain shader types.  It's ugly, but necessary without refactoring other code.